### ISSUE TITLE
feature: Add script to split generated clients to their own commit and update docs

### ISSUE THREAD

**Issue by DerekStrickland** [hashicorp/nomad-openapi#38]
- [ ] Split client directory changes into a separate commit automatically
- [ ] Investigate if it could be a commit hook
- [ ] Investigate if it should create one commit per language

**Comment by Neutrollized** [comment]
I don't think think can be implemented as a pre-commit hook.  As that operates on stashed items...unless the workflow happens like:
1) `git reset`
2) loop the following for each client directory: add dir to add/stash, commit AND push with a templated commit message
3) finally will add/stash the remaining code and commit

**Comment by Neutrollized** [comment]
Ooops! I meant staged, not stashed.

**Comment by DerekStrickland** [comment]
Yeah I think I get it now. I suppose you could get clever and write a post commit hook but that feels dangerous 🤔 

Maybe the solutions is to have the `make v1` command just commit the clients folder then, and if the author runs it multiple times, it's on them or the reviewer to squash those?  It could move things forward...

**Comment by Neutrollized** [comment]
I've been playing with some stuff and probably at the next meeting I can show what I have and we can discuss how best to approach this.  I don't think there's any ideal solution -- there will be some compromise either way, it's just a question of which one we think is the easiest to implement.

**Comment by DerekStrickland** [comment]
Sounds good!  Looking forward to it.
